### PR TITLE
add containerfile

### PR DESCRIPTION
**Pull request by ucfchandra**
Containerfile is written here and works. The commands to test it are:

`podman machine init`

`podman machine start`

`podman build -f Containerfile-rc -t announcements .`

and then

`podman run -ti -p 8000:8000 announcements`

Minor updates were necessary for the pip requirements due to the upgrade from python 2 to 3.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,3 +29,5 @@ gulp-config.json
 # Static Files
 static/admin/*
 static/rest_framework/*
+
+.odo
```
Comment (jmbarne3) [inline .gitignore:33]: Add a trailing newline to the end of the file, please.

File: devfile.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+commands:
```
Comment (jmbarne3) [inline devfile.yaml:1]: What is this file and what is it doing? 😅

Should there be additional readme documentation explaining what this is and how to use it? Is this an optional component/tool that is available for developers, but not necessary to run the project?

File: Containerfile-rc [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#Windows/Linux
+#FROM python:3.12.1-bookworm
+
+#Mac
+FROM amd64/python:3.12.1-bookworm
+
+#Mac M1
+#FROM arm64v8/python:latest
```
Comment (jmbarne3) [inline Containerfile-rc:8]: We might consider trying multiple `FROM` statements to build multiple images, one for each build system we need to support. See documentation on multi-platform images: https://docs.docker.com/build/building/multi-platform/.

**Review by jmbarne3 — APPROVED** [review]
Looks good. Please provide some feedback on the devfile question and update the readme as appropriate.

File: devfile.yaml (hunk 2 above)
Comment (ucfchandra) [reply devfile.yaml:1]: Sorry, this should be in the odo branch. It's not doing anything. Anything odo-related was accidentally pushed into this, I'll remove that stuff, it's not relevant to this PR

**Review by kimberspencer — APPROVED** [review]

**Comment by Ramin-Farhadi** [comment]
1. Add the brew install podman to the readme then the following commends 

podman machine init
podman machine start
podman build -f Containerfile-rc -t announcements .

Everything works fine for me. Since there was another README in the announcements application, it was a little confusing at first. I suggest adding the instructions in a different README file or Wiki.

**Review by Ramin-Farhadi — APPROVED** [review]

**Pull request closed without merging**